freud/data.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2010-2019 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+R"""
+The :class:`freud.data` module provides certain sample data sets and utility
+functions that are useful for testing and examples.
+"""
+
+import numpy as np
+import freud
+
+
+class UnitCell(object):
+    """Class to represent the unit cell of a crystal structure.
+
+    This class represents the unit cell of a crystal structure, which is
+    defined by a lattice and a basis. It provides the basic attributes of the
+    unit cell as well as enabling the generation of systems of points
+    (optionally with some noise) from the unit cell.
+
+    Args:
+        box:
+            A box-like object (see :meth:`~freud.box.Box.from_box`) containing
+            the lattice vectors of the unit cell.
+        basis_positions ((:math:`N_{points}`) :class:`numpy.ndarray`):
+            The basis of the unit cell in fractional coordinates
+            (Default value = ``[[0, 0, 0]]``.
+    """
+    def __init__(self, box, basis_positions=[[0, 0, 0]]):
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = np.dot(basis_positions, self.box.to_matrix().T)
```

Comment:
This is misleading to accept an argument in fractional coordinates and return a property in real coordinates. Also, this should definitely use the box's `make_fractional` and `make_absolute` methods.

Reply:
Agreed on all counts.